sound/soc/sof/sof-audio.c

```diff
@@ -107,6 +107,17 @@ int sof_widget_free(struct snd_sof_dev *sdev, struct snd_sof_widget *swidget)
 	if (!swidget->private)
 		return 0;
 
+	/*
+	 * PCM streams can be freed without being restarted after system resume. Prevent
+	 * the use_count from being decremented below 0 for dynamic widgets and below 1
+	 * for static widgets in such scenarios.
+	 */
+	if (swidget->dynamic_pipeline_widget && !swidget->use_count)
+			return 0;
+
+	if (!swidget->dynamic_pipeline_widget && swidget->use_count == 1)
```

Comment:
I personally don't get why the use count is different for dynamic and static pipelines. A comment would not hurt...